### PR TITLE
Simplify build jobs using a matrix

### PR DESCRIPTION
**Pull request by saurabhyadavdev**
Motivation: 
Reduce duplication in the GitHub Actions workflow unit-tests https://github.com/asyncer-io/r2dbc-mysql/issues/83

Modification: 
Refactored three jobs into one, now will build using strategy.matrix.version

Result: 
Jobs should build for Java 8, 11 and 17 version.

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -3,39 +3,18 @@ name: Unit tests
 on: [ pull_request ]
 
 jobs:
-  build_java_8:
+  build:
```
Comment (jchrys) [inline .github/workflows/unit-tests.yml]: Could you modify the code to clarify its purpose?
```suggestion
  linux-build-java-${{ matrix.version }}:
```

**Review by jchrys — CHANGES_REQUESTED** [review]
Thank you very much for your PR! Just one small thing.

+ Could you please rebase and squash the changes to create a single commit?

Could you please add a line separator in the commit message, as shown below, to improve readability?
```
Motivation: 
Reduce ...

Modification: 
Refactored...

Result: 
Jobs...
```

**Comment by saurabhyadavdev** [comment]
Hey can you please delete my fork so I can fork again and raise PR as I am unable to squash into one commit

**Comment by jchrys** [comment]
You don't need to delete your fork to achieve this; you can do it by yourself.

Try the following steps:

$ git checkout {yourbranch}
$ git rebase -i trunk
Squash all commits except the top one.
Create a commit message.
$ git push --force

**Review by jchrys — APPROVED** [review]

**Comment by jchrys** [comment]
I squash merged to make it a single commit.
Thanks a lot!

**Pull request merged**